service/matching/matching_engine.go

```diff
@@ -569,11 +594,67 @@ pollLoop:
 
 			continue pollLoop
 		}
+
 		task.finish(nil)
 		return e.createPollWorkflowTaskQueueResponse(task, resp, opMetrics), nil
 	}
 }
 
+// getHistoryForPollWorkflowTaskQueueQueryTask retrieves history associated with a query task returned
+// by PollWorkflowTaskQueue. Returns empty history for sticky query and full history for non-sticky
+func (e *matchingEngineImpl) getHistoryForPollWorkflowTaskQueueQueryTask(
+	ctx context.Context,
+	nsID namespace.ID,
+	task *internalTask,
+	isStickyEnabled bool,
+) (*history.History, []byte, error) {
+	if isStickyEnabled {
+		return &history.History{Events: []*history.HistoryEvent{}}, nil, nil
+	}
+
+	continuation := &tokenspb.HistoryContinuation{
+		RunId:        task.workflowExecution().GetRunId(),
+		FirstEventId: common.FirstEventID,
+	}
+
+	continueToken, err := api.SerializeHistoryToken(continuation)
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
I don't think this is needed and matching is not supposed to construct this token after all old code is deprecated.